namerd/storage/consul/src/main/scala/io/buoyant/namerd/storage/consul/ConsulDtabStore.scala

```diff
@@ -38,19 +43,20 @@ class ConsulDtabStore(
               case Return(result) =>
                 val namespaces = result.value.map(namespace).toSet
                 updates() = Activity.Ok(namespaces)
-                cycle(result.index)
+                cycle(result.index, backoffs)
               case Throw(e: NotFound) =>
                 updates() = Activity.Ok(Set.empty[Ns])
-                cycle(e.rsp.headerMap.get(Headers.Index))
+                cycle(e.rsp.headerMap.get(Headers.Index), backoffs)
               case Throw(e: Failure) if e.isFlagged(Failure.Interrupted) => Future.Done
               case Throw(e) =>
                 updates() = Activity.Failed(e)
                 log.error("consul ns list observation error %s", e)
-                cycle(None)
+                val sleep #:: backoffs0 = backoffs
+                Future.sleep(sleep).flatMap(_ => cycle(None, backoffs0))
```

Comment:
TIOLI: flatMap is perfectly fine here but there is also `Future.sleep(sleep).before(cycle(None, backoffs0))` that is equivalent.  Take your pick.